src/opendr/perception/object_detection_2d/utils/class_filter_wrapper.py

```diff
@@ -0,0 +1,137 @@
+from opendr.perception.object_detection_2d.retinaface.retinaface_learner import RetinaFaceLearner
+from opendr.perception.object_detection_2d.ssd.ssd_learner import SingleShotDetectorLearner
+from opendr.perception.object_detection_2d.yolov3.yolov3_learner import YOLOv3DetectorLearner
+from opendr.perception.object_detection_2d.yolov5.yolov5_learner import YOLOv5DetectorLearner
+from opendr.perception.object_detection_2d.nanodet.nanodet_learner import NanodetLearner
+
+
+class FilteredLearnerWrapper:
+    def __init__(self, learner, allowed_classes=None):
+        self.learner = learner
+        self.allowed_classes = allowed_classes if allowed_classes is not None else []
+
+        if isinstance(self.learner,
+                      (CenterNetDetectorLearner, YOLOv3DetectorLearner, YOLOv5DetectorLearner, NanodetLearner,
+                       RetinaFaceLearner, SingleShotDetectorLearner)):
+            self.classes = self.learner.classes
+        if isinstance(self.learner, DetrLearner):
+            coco_classes = [
+                "N/A", "person", "bicycle", "car", "motorcycle", "airplane", "bus", "train", "truck", "boat",
+                "traffic light", "fire hydrant", "N/A", "stop sign", "parking meter", "bench", "bird", "cat", "dog",
+                "horse", "sheep", "cow", "elephant", "bear", "zebra", "giraffe", "N/A", "backpack", "umbrella", "N/A",
+                "N/A", "handbag", "tie", "suitcase", "frisbee", "skis", "snowboard", "sports ball", "kite",
+                "baseball bat", "baseball glove", "skateboard", "surfboard", "tennis racket", "bottle", "N/A",
+                "wine glass", "cup", "fork", "knife", "spoon", "bowl", "banana", "apple", "sandwich", "orange",
+                "broccoli", "carrot", "hot dog", "pizza", "donut", "cake", "chair", "couch", "potted plant", "bed",
+                "N/A", "dining table", "N/A", "N/A", "toilet", "N/A", "tv", "laptop", "mouse", "remote", "keyboard",
+                "cell phone", "microwave", "oven", "toaster", "sink", "refrigerator", "N/A", "book", "clock", "vase",
+                "scissors", "teddy bear", "hair drier", "toothbrush",
+            ]
+            self.classes = coco_classes
+
+        # Verify that allowed classes are in the detector's class list
+        invalid_classes = [cls for cls in self.allowed_classes if cls not in self.classes]
+        if invalid_classes:
+            raise ValueError(
+                f"The following classes are not detected by this detector: {', '.join(invalid_classes)}")
+
+    def infer(self, img=None, threshold=None, keep_size=None, input=None, conf_threshold=None, iou_threshold=None,
+              nms_max_num=None, nms_threshold=None, scales=None, mask_thresh=None, size=None, custom_nms=None,
+              nms_thresh=None, nms_topk=None, post_nms=None, extract_maps=None):
+
+        # match variable names
+        if isinstance(self.learner, NanodetLearner):
+            img = input
```

Comment:
For the wrapper to serve as a drop in replacement, we should assume that infer will be used without a keyword argument, e.g. in the nanodet webcam demo:

`boxes = filtered_nanodet.infer(img, conf_threshold=0.35, iou_threshold=0.6, nms_max_num=20)`

This leads to `input=None` which is then assigned to img. I suggest to add another check:

```suggestion
        if isinstance(self.learner, NanodetLearner) and input is not None:
            img = input
```

And change the infer to use `img` instead (next suggestion).